### ISSUE TITLE
Warning Elimination: ignoring attributes in template argument, emitted by GCC

### ISSUE THREAD

**Issue by MarshallW906** [gore-engine/gore#23]
This is because types like `__m128` have attributes when they are defined (see below), which will be ignored when passed as template arguments

```
// /usr/lib/gcc/x86_64-linux-gnu/11/include/xmmintrin.h
typedef float __m128 __attribute__ ((__vector_size__ (16), __may_alias__));
```

**Comment by MarshallW906** [comment]
TL;DR: Don't passing types as template arguments when the types have special attributes applied during typedef.

Okay this means I must roll back and manually implement the two constructors `InterfaceType(const SIMDValue&)` and `InterfaceType(SIMDValue&&)`, instead of using perfect forwarding.

Below are the reason:
This has been a known issue for years, and it can cause sizeof/alignment problems. 

Actually both GCC and Clang have this problem, while only GCC reports it. 
- https://gcc.gnu.org/bugzilla/show_bug.cgi?id=97222
- https://bugs.llvm.org/show_bug.cgi?id=47674

MSVC seems fine but I can't confirm. It behaves as expected on alignment, but the `xmmintrin.h` provided in Windows SDK does not depends on a `vector_size()` attribute (they use union and floats), which is used in X11's `xmmintrin.h`
